src/DonationForms/resources/registrars/templates/fields/Amount/useCurrencySeparators.tsx

```diff
@@ -0,0 +1,26 @@
+import { useState, useEffect } from 'react';
+
+/**
+ * Custom hook to determine the group and decimal separators based on locale.
+ */
+export default function useCurrencySeparator(){
+    const [groupSeparator, setGroupSeparator] = useState(',');
+    const [decimalSeparator, setDecimalSeparator] = useState('.');
+
+    useEffect(() => {
+        const formatter = new Intl.NumberFormat();
```

Comment:
This will override the browser's locale and assume a US locale for separators.

While is does prevent the `NaN` issue, we shouldn't do so by locking the local config.

Reply:
Also, I don't think we need a hook for this, since the value shouldn't change after page load. Instead, we could initialize these variables at the top of the component file,

```tsx
import ...

const formatter = new Intl.NumberFormat();
// ...

const Component = () => {
    // ...
}

export default Component;
```

Reply:
@JoshuaHungDinh I updated this to use the browser locale configuration and replaced the non-breaking space on the group separator to avoid any conflicts with the suffix separator used by the `<CurrencyInput />` component.

```tsx
const formatter = new Intl.NumberFormat(navigator.language);
const groupSeparator = formatter.format(1000).replace(/[0-9]/g, '');
const decimalSeparator = formatter.format(1.1).replace(/[0-9]/g, '');

  //
  groupSeparator={
    /**
     * Replace non-breaking space to avoid conflict with the suffix separator.
     * @link https://github.com/cchanxzy/react-currency-input-field/issues/266
     */
    groupSeparator.replace(/\u00A0/g, ' ')
  }
  //
```

Reply:
Awesome, that makes sense.

Thanks for helping me move this forward. I'll send it to QA.